cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -888,6 +891,38 @@ public List<ProgramHistory> getRuns(Collection<ProgramId> programs, ProgramRunSt
     });
   }
 
+  @Override
+  public Optional<byte[]> getState(AppState request) {
+    return TransactionRunners.run(transactionRunner, context -> {
+      return getAppStateTable(context).getState(request);
```

Comment:
inside the transaction, we need to check that the app exists and throw ApplicationNotFoundException if not.